### ISSUE TITLE
Error: Template parse errors

### ISSUE THREAD

**Issue by chrisj-skinner** [getsaf/shallow-render#81]
I get the following error when trying to apply the demo @ https://github.com/getsaf/shallow-render/wiki into my app.

Error: Template parse errors:​​
    ​​Unexpected closing tag "__MockOfEmptyOutletComponent-selector". It may happen when the tag has already been closed by another tag. For more info see https://www.w3.org/TR/html5/syntax.html#closing-elements-that-have-implied-end-tags ("<__MockOfEmptyOutletComponent-selector>[ERROR ->]</__MockOfEmptyOutletComponent-selector>"): @0:39​​

**Comment by getsaf** [comment]
Thanks @mihalcan, I was actually looking into this today, and I found the underlying issue.

There's a feature that I have while mocking components that attempts to "correct" for components that do not have a selector defined (because the TestBed JIT Compiler bombs when it sees such a component).

[Here's the function](https://github.com/getsaf/shallow-render/blob/master/lib/tools/ng-mock.ts#L12-L17) that applies the "correction".

It looks like the selector I used to correct with was a tag selector and apparently, that breaks the `EmptyOutletComponent` mocking because I *think* it's trying to generate an open and close tag for the component dynamically. I switched the selector to a css-class selector (prefixing it with a `.`) and it seems to have solved that template parse error!!! Wooo!!!! This has been bugging folks for a while! I've got a PR (#82) up with the fix.

[Here's the line that fixes it!](https://github.com/getsaf/shallow-render/pull/82/files#diff-15613285ea2234b374adec0187816abcL15)

I should note that just because this bypasses that explosion, it's still recommended that you use the `RouterTestingModule` in your tests. This fix will just address the red-herring error message that is currently seen when mocking the `RouterModule` directly.
